Templates/DefaultProject/Template/package.bat

```diff
@@ -0,0 +1,16 @@
+REM --------------------------------------------------------------------------------------------------
+REM 
+REM Copyright (c) Contributors to the Open 3D Engine Project.
+REM For complete copyright and license terms please see the LICENSE at the root of this distribution.
+REM 
+REM SPDX-License-Identifier: Apache-2.0 OR MIT
+REM 
+REM --------------------------------------------------------------------------------------------------
+
+set O3DE_PATH=${EnginePath}
+set O3DE_PROJECT_PATH=${ProjectPath}
```

Comment:
We talked about maybe using %~dp0 for this so the project was portable, but then the engine path would need to be adjusted too so maybe it'd be better to just add some comments in this script saying the user may need to update the paths.  Also I wonder if we should check the paths exist so we can give a nicer error message if you download someone else's project and try to run the script.

Reply:
Or even have the cmake project generation also generate this script, but then that gets really tricky. I'll use %~dp0 instead

Reply:
this is what you get if you run `cmake -P cmake/EngineFinder.cmake` from the project folder
```bash
λ cmake -P cmake\EngineFinder.cmake
-- Selecting engine 'C:/O3DE/23.10.1' from 'engine_path' in '<project>/user/project.json'.
```
most of the engine path output text from that script is in the pattern `Selecting engine <engine path> ...` except the unusual case where it is provided by the CMAKE_MODULE_PATH.  not sure we want to go down this road but that is one way to get the engine path